pkg/scheduler/objects/queue.go

```diff
@@ -628,6 +635,8 @@ func (sq *Queue) AddApplication(app *Application) {
 	defer sq.Unlock()
 	sq.applications[app.ApplicationID] = app
 	sq.appPriorities[app.ApplicationID] = app.GetAskMaxPriority()
+	sq.appPreemptionPriorities[app.ApplicationID] = app.GetAllocationMinPriority()
+
```

Comment:
I did not see this when we added priorities earlier: when an application gets added to a queue there are no asks or allocations yet. The `queue.AddApplication()` call is triggered from a new application request from a shim and that request cannot have an ask/allocation. The init can be a simple default value for both.

Reply:
Technically it's true, I still prefer this approach because it slightly reduces coupling. cc @craigcondit

Reply:
The application object cannot be accessed by anything until the end of this function itself.  The queue is locked.
The overhead is 2 function calls, and 2 lock acquire and release actions. Locking cannot be contested as the application object cannot be accessed yet. That overhead is less than the Tag check...
OK to leave it.